### ISSUE TITLE
App crashes on iOS 10 permission dialog --> undefined

### ISSUE THREAD

**Issue by patrickhofer** [phonegap/phonegap-plugin-barcodescanner#329]
### Expected Behaviour
Request permission for camera and open the camera

### Actual Behaviour
First time launch iOS aks for permisson. But the Dialog shows a new undefined string which was not there in iOS 9. The error that the device has no access to the camera fires and it's not possible at all cost to start the camera until you kill the app and fire the camera again. Then everthing works fine. The permission to use the camera doesn't work on iOS 10.0. 

### Reproduce Scenario (including but not limited to)
cordova 6.3.1
ionic 2.1.0-beta.1
OS X El Captain

Install an app with the plugin on a iOS 10 device with the following code: 

```
$scope.scan = function() {
      $cordovaBarcodeScanner.scan().then(function(barcodeData) {
        if (barcodeData.text === "mappar1_ital") {
          save("isUnlocked", true);

          if ($state.current.name === "l1") {
            $state.go("11");
          }
          if ($state.current.name === "l2") {
            $state.go("12");
          }
        }

      }, function(error) {
        console.log(JSON.stringify(error));
      });

      function alertDismissed() {
        console.log("OK");
      }

      cordova.plugins.diagnostic.isCameraAuthorized(function(authorized) {
        if (authorized === false) {
          navigator.notification.alert(
            'Bitte aktivieren Sie den Zugriff auf die Kamera in den Einstellungen!',
            alertDismissed, // message
            'Kein Zugriff', // title
            'OK' // buttonName
          );
        }
      }, function(error) {
        console.error("The following error occurred: " + error);
      });
    }
```

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)

ios 10.0 

#### Plugin version

   phonegap-plugin-barcodescanner newest

Any help much appreciated!

**Comment by jcesarmobile** [comment]
Which plugin version are you using? this has been fixed

**Comment by patrickhofer** [comment]
I've deinstalled the plugin and added it again with `ionic plugin add phonegap-plugin-barcodescanner`

**Comment by jcesarmobile** [comment]
Are you getting version 6.0.2?
That's the one with the fix

To check the version use `ionic plugin list`

**Comment by patrickhofer** [comment]
`phonegap-plugin-barcodescanner 6.0.2 "BarcodeScanner"` 

Yes, 6.0.2

**Comment by jcesarmobile** [comment]
The app being blocked is probably because the app ask for the permission and you show your alert over it, don't do it. The plugin will check if the permission is set or not and show a message.

But the "undefined" string shouldn't be shown, it should be an empty string

**Comment by patrickhofer** [comment]
Thank you! Can you tell me how I can remove the undefined in the permission alert? We will go into production in 5 days =)

**Comment by jcesarmobile** [comment]
You can install with 
`ionic plugin add phonegap-plugin-barcodescanner --variable CAMERA_USAGE_DESCRIPTION="the reason you want to give to scan barcodes"` or `ionic plugin add phonegap-plugin-barcodescanner --variable CAMERA_USAGE_DESCRIPTION=" "`

**Comment by Cordobo** [comment]
I experienced a similar problem - I had to update the App to support iOS 10 (which requires a NSCameraUsageDescription), but couldn't upgrade the barcodescannerplugin from 4.x.x to V. 6.x.x, so I wrote this plugin.

**NPM**
`cordova plugin add cordova-plugin-ios-camera-permissions --save`

**GutHub Repo**
https://github.com/Cordobo/cordova-plugin-ios-camera-permissions

HTH

**Comment by jcesarmobile** [comment]
You can also install cordova-plugin-camera 

`cordova plugin add cordova-plugin-camera --variable CAMERA_USAGE_DESCRIPTION="the reason you want to give to scan barcodes"`

**Comment by Cordobo** [comment]
@jcesarmobile True, for the latest version. Unfortunatelly I couldn't update to 6.x.x, and as it's not working with  4.2.1, I wrote this mini-plugin.

I even tried this install routine on 4.x.x but with no success :)

**Comment by jcuervo** [comment]
Is there any reason why permissions alert doesn't show up promptly? I'm testing on iOS 10 now and it's very frustrating.

**Comment by Cordobo** [comment]
@jcuervo here they come up the first time the app requests the permissions. Maybe you can trigger them in the first view/controller.

**Comment by jcuervo** [comment]
I'll try it out. Thanks, buddy!

**Comment by jcesarmobile** [comment]
@Cordobo I mean't installing the camera plugin instead of updating the barcode plugin

@jcuervo the permissions should appear when you are going to use that permission

**Comment by jcuervo** [comment]
@jcesarmobile i can now use the barcode scanner, but I have to close the app (tap the home button) and tap the app again then the barcode scanner appears. you ever encountered this? (neer had this issue on previous iOS versions)

**Comment by jcesarmobile** [comment]
I have not encountered this issue myself, but I think there is an issue open about it, usually when using phonegap build

**Comment by jcuervo** [comment]
@jcesarmobile thanks for the info! btw, is this related to the camera plugin or on cordova itself?

**Comment by jcesarmobile** [comment]
@jcuervo no idea

**Comment by asierazaceta** [comment]
My app was not working on iOS10 with latest phonegap barcode scanner plugin (6.0.3) until I found this post related to the CSP, which fixed it by adding the gap: to the allowed content:
[http://stackoverflow.com/a/39638649/1752548](http://stackoverflow.com/a/39638649/1752548)

**Comment by jcuervo** [comment]
@asierazaceta wow! it worked! thanks, man!
